aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -13,25 +13,42 @@ namespace {
 
 using namespace vec256;
 
-void add_kernel(TensorIterator& iter, Scalar alpha_scalar) {
+// `add` tells us whether this function is called from add (true) or sub (false), so we can handle
+// some corner cases, such as bool type and dispatch name. Note that it is always a + b * alpha that
+// is performed, regardless of what `add` is set to.
+template <bool add>
+inline void add_kernel_impl(TensorIterator& iter, Scalar alpha_scalar) {
   if (iter.dtype() == ScalarType::Bool) {
-    auto alpha = alpha_scalar.to<bool>();
-    cpu_kernel(iter, [=](bool a, bool b) -> bool { return a + b * alpha; });
+    if (add) {
+      auto alpha = alpha_scalar.to<bool>();
+      cpu_kernel(iter, [=](bool a, bool b) -> bool { return a + b * alpha; });
+    } else {
+      AT_ERROR("'sub' does not support bool type.");
```

Comment:
ideally there should be two cases: 
1) boolTensor - boolTensro
Should give an error like "'sub_cpu' is not supported for boolean tensors" or similar.

2) 1 - boolTensor
Should give an error like "This behavior is about to change. Please use '^' instead" or similar.

Reply:
@gchanan, do you agree with proposed error messages?

Reply:
> ideally there should be two cases:
> 
>     2. 1 - boolTensor
>        Should give an error like "This behavior is about to change. Please use '^' instead" or similar.

How do we know the first operand is 1?

Reply:
For 1, numpy's error message seems reasonable (except for deprecation):

```
numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
```
which of these do we actually support?

For 2), I think this should be handled by @nairbv's PR.

Reply:
Also, I don't think there's any benefit to specifying cpu/cuda in the error message name, both cpu and cuda have the same behavior.

Reply:
> For 1, numpy's error message seems reasonable (except for deprecation):
> 
> ```
> numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
> ```
>

Good point.
 
> which of these do we actually support?

Looks like none...

Reply:
> How do we know the first operand is 1?

@xuhdev: I don't think that's something you need to worry about in this PR, that should be handled separately in https://github.com/pytorch/pytorch/pull/22273.

Reply:
> Looks like none...

ouch, that's not good.

Reply:
We support '^'
```

import torch

a = torch.tensor([1, 0, 1, 0, 1, 0], dtype=torch.uint8)
b = torch.tensor([0, 1, 0, 1, 0, 1], dtype=torch.uint8)

print(a - b)
print(a ^ b)

print("\n")
print(a.bool() - b.bool())
print(a.bool() ^ b.bool())
```

res: 
```
tensor([  1, 255,   1, 255,   1, 255], dtype=torch.uint8)
tensor([1, 1, 1, 1, 1, 1], dtype=torch.uint8)


tensor([True, True, True, True, True, True])
tensor([True, True, True, True, True, True])
```

Reply:
actually it looks like we support `^`, see:
```
>>> torch.tensor([True, True, False, False]) ^ torch.tensor([True, False, True, False])
tensor([False,  True,  True, False])
```

Reply:
given that, I'd suggest the error message:
```
torch boolean subtraction, the `-` operator, is not supported, use the `^` operator instead.
```